runtime/src/registry.rs

```diff
@@ -274,24 +274,19 @@ decl_module! {
 
         #[weight = SimpleDispatchInfo::InsecureFreeNormal]
         pub fn unregister_user(origin, message: message::UnregisterUser) -> DispatchResult {
-            fn can_be_unregistered(user: state::User, sender: AccountId) -> bool {
-                user.account_id == sender && user.projects.is_empty()
-            }
-
             let sender = ensure_signed(origin)?;
+            let sender_user = get_user_with_account(sender).ok_or(RegistryError::InexistentUser)?;
 
-            match store::Users::get(message.user_id.clone()) {
-                None => Err(RegistryError::InexistentUser.into()),
-                Some(user) => {
-                    if can_be_unregistered(user, sender) {
-                        store::Users::remove(message.user_id.clone());
-                        Self::deposit_event(Event::UserUnregistered(message.user_id));
-                        Ok(())
-                    }
-                    else {
-                        Err(RegistryError::UnregisterableUser.into())
-                    }
-                }
+            if sender_user.id != message.user_id {
+                Err(RegistryError::InsufficientSenderPermissions.into())
+            }
+            else if find_org(&|org| org.members.contains(&sender_user.id)).is_some() {
```

Comment:
```suggestion
            } else if find_org(&|org| org.members.contains(&sender_user.id)).is_some() {
```

This formatting looks off.

Reply:
Seems strange to have this extra function just to call a couple of iterator methods.

Reply:
> Seems strange to have this extra function just to call a couple of iterator methods.

yeah. I just wanted to make the general logic reusable while still making it declarative and readable. we can go yagni style but it doesn't hurt at all to have it ready for reuse.